R/OsModel.R

```diff
@@ -0,0 +1,118 @@
+#' Title - TODO
+#'
+#' Description - TODO
+#'
+#' @slot stan TODO
+#' @slot templated Logical. Defines whether the OsModel object needs to be completed
+#' @export
+OsModel <- setClass(
+    Class = "OsModel",
+    representation = list(
+        stan = "StanModule"
+    )
+)
+
+
+
+#' LogLogisticModule
+#'
+#' Log logistic module helper function.
+#' @param functions A stan code including the functions section of the model.
+#' @param data  A stan code including the data section of the model.
+#' @param parameters  A stan code including the parameters section of the model.
+#' @param transformed_parameters  A stan code including the transformed parameters section of the model.
+#' @param generated_quantities  A stan code including the generated quantities section of the model.
+#' @param priors  A prior list with the priors of the model.
+#' @param model  A prior list with the priors of the model.
+#' @param inits  A list with the initial values.
+#' @export
+LogLogisticModule <- function(functions = "os_functions.stan",
+                              data = "os_data.stan",
+                              parameters = "os_parameters.stan",
+                              transformed_parameters = "os_transformed_parameters.stan",
+                              generated_quantities = "os_generated_quantities.stan",
+                              priors = os_prior(),
+                              model = "target += sum(log_lik);",
+                              inits = list()) {
+    StanModule(
+        functions = functions,
+        data = data,
+        parameters = parameters,
+        transformed_parameters = transformed_parameters,
+        generated_quantities = generated_quantities,
+        priors = priors,
+        model = model,
+        inits = inits
+    )
+}
+
+
+
+
+
+
+
+#' Parametrize TemplatedStanOs object with the selected Hazardlink
+#' @param osmod TemplatedStanOs object
+#' @param link HazardLink object
+#' @export
+setGeneric("parametrize", function(osmod, link) {
+    standardGeneric("parametrize")
+})
+
+
+
+setMethod(
```

Comment:
where is this documented? I would suggest to use `@rdName parametrize` to put this in the same place as the generic function